File: templates/ios/movin-example-app/BaseViewController.m
//
//  BaseViewController.m
//  movin-example-app
//
//  Created by Tim Potze on 28/06/16.
//  Copyright © 2016 Movin. All rights reserved.
//

#import "BaseViewController.h"

@implementation BaseViewController

// A helper method which creates an UIAlertController with an UIAlertControllerStyleAlert style with the specified
// title and message.
- (void)alertWithTitle:(NSString*)title
            andMessage:(NSString*)message
           andCanClose:(BOOL)canClose {

    // This block will be executed on the main thread.
    void (^showMessage)() = ^{
        // Create the alert controller.
        UIAlertController* alert = [UIAlertController alertControllerWithTitle:title
                                                                       message:message
                                                                preferredStyle:UIAlertControllerStyleAlert];

        // Add a single action to the alert controller if the alert can be closed, the OK button.
        if(canClose) {
            UIAlertAction* ok = [UIAlertAction
                    actionWithTitle:@"OK"
                              style:UIAlertActionStyleDefault
                            handler:^(UIAlertAction* action) {
                                [alert dismissViewControllerAnimated:YES completion:nil];
                            }];

            [alert addAction:ok];
        }
        // Present the alert.
        [self presentViewController:alert animated:YES completion:nil];
    };

    if ([NSThread isMainThread]) {
        showMessage();
    }
    else {
        dispatch_sync(dispatch_get_main_queue(), showMessage);
    }
}

@end

File: templates/ios/movin-example-app-swift/BaseViewController.swift
//
//  BaseViewController.swift
//  movin-example-app-swift
//
//  Created by Tim Potze on 18/07/16.
//  Copyright © 2016 Movin. All rights reserved.
//

import UIKit

class BaseViewController: UIViewController {
    func alert(title: String, message: String, canClose: Bool) {
        func showMessage() -> () {
            let alert = UIAlertController(title: title, message: message, preferredStyle: UIAlertControllerStyle.alert)

            if canClose {
                let ok = UIAlertAction(title: "OK", style: UIAlertActionStyle.default) {
                    (action) -> () in
                    alert.dismiss(animated: true, completion: nil)
                }

                alert.addAction(ok)
            }

            self.present(alert, animated: true, completion: nil)
        }

        if Thread.isMainThread {
            showMessage();
        } else {
            DispatchQueue.main.async(execute: showMessage)
        }
    }
}
